pyxcp/transport/base.py

```diff
@@ -218,5 +256,6 @@ def processResponse(self, response, length, counter):
         else:
             if self.first_daq_timestamp is None:
                 self.first_daq_timestamp = datetime.now()
-            # self.daqQueue.put((response, counter, length))
-            self.daqQueue.append((response, counter, length))
+            timestamp = perf_counter()
```

Comment:
Did you measure the impact of always calling perf_counter and creating the daqeleemnt namedtuple?

Reply:
Well, I did it now.

```python
import time
from typing import NamedTuple
from collections import namedtuple

MySimpleNamedTuple = namedtuple('MySimpleNamedTuple', ['payload', 'foo', 'bar', 'timestamp'])

class MyNamedTuple(NamedTuple):
    payload: bytes
    foo: int
    bar: int
    timestamp: float

def create_tuple():
    pl = bytes([1,2,3,4,5,6,7,8])
    t = (pl, 1, 2, time.perf_counter())
    
def create_namedtuple():
    pl = bytes([1,2,3,4,5,6,7,8])
    nt = MyNamedTuple(pl, 1, 2, time.perf_counter())
    
def create_simple_namedtuple():
    pl = bytes([1,2,3,4,5,6,7,8])
    nt = MySimpleNamedTuple(pl, 1, 2, time.perf_counter())
    
>> %timeit create_tuple()
>> 443 ns ± 3.04 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

>> %timeit create_namedtuple()
>> 753 ns ± 6.12 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

>> %timeit create_simple_namedtuple()
>> 741 ns ± 3.2 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

So the NamedTuple creation is about half as fast... I think I'll undo it, it also does not work this way under py3.6.

Calling `time.perf_counter()` takes ~100ns:
```python
>> %timeit time.perf_counter()
>> 102 ns ± 0.215 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```
but maybe it should only be called if the slave device actually does not support timestamping.

Reply:
> but maybe it should only be called if the slave device actually does not support timestamping.

I think so too

Reply:
Well,  your (CAN)  interface may also support timestamping, so a parameter could be the best choice? 

Btw,  I also did some (more or less off-topic)  benchmarking:
1) when it comes to req/resp communication, the performance seems to be limited by the OS overhead/latency in the first place.  I'd measured the upload speed for CTO sizes between 8 and 64 bytes (code will be available under *examples/xcp_read_bechmark. py*, after I merged your PR):
![ReqRespPerformance](https://user-images.githubusercontent.com/10755523/63754950-054c4b80-c8b6-11e9-9d3b-cc5906f4c4b2.png)
2. *time.time()* seems to be a bit faster than *time. perf_counter()*,  but note:
*perf_counter()* uses the most precise timing facility available on your OS (QueryPerformanceCounter() on Windows),  making it the natural choice for timestamping.

Reply:
Sorry, but I think namedtuples are not that useful in this situation -- I just removed DaqQueueElement.

Reply:
Yeah, I just missed your merge, I already removed it as well. Also, added option for timestamping, based on danielhrisca.

Reply:
I have created #35 containing the optional timestamping.

Reply:
Hi Waszil, 
one rollback again... regarding the Can-constructor.

Reply:
PS. :  I'm using Kvaser SDK to run virtual CAN channels between pyxcp and cxcp.

Reply:
But wait,  I'm not sure if passing a class is a better idea than passing an instance....

Reply:
Hi, well, in my case actually, the canInterface instance is already created elsewhere, so passing a class does not work for me, but that's my problem. However, I think, as there is the abstract base class, it suggests, that any specific implementation can exist, and in this case, passing a class prevents these instances being created elsewhere.

Reply:
Oh, and about the timestamping switch: I created the enable/disable methods so that this feature can be switched on/off runtime, when the master actually gets the information from the slave device via get_daq_processor_info. I'm not sure that this should be preset via the configuration